extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts

```diff
@@ -381,16 +387,142 @@ describe("db panel", () => {
       dateAdded: 1668428293677,
       language: "csharp",
       storagePath: "/path/to/db1/",
+      selected: false,
     });
     checkLocalDatabaseItem(localDatabaseItems[1], {
       kind: DbItemKind.LocalDatabase,
       databaseName: "db2",
       dateAdded: 1668428472731,
       language: "go",
       storagePath: "/path/to/db2/",
+      selected: false,
     });
   });
 
+  it("should mark selected remote db list as selected", async () => {
+    const dbConfig: DbConfig = {
+      databases: {
+        remote: {
+          repositoryLists: [
+            {
+              name: "my-list-1",
+              repositories: ["owner1/repo1", "owner1/repo2"],
+            },
+            {
+              name: "my-list-2",
+              repositories: ["owner2/repo1", "owner2/repo2"],
+            },
+          ],
+          owners: [],
+          repositories: [],
+        },
+        local: {
+          lists: [],
+          databases: [],
+        },
+      },
+      selected: {
+        kind: SelectedDbItemKind.RemoteUserDefinedList,
+        listName: "my-list-2",
+      },
+    };
+
+    await saveDbConfig(dbConfig);
+
+    const dbTreeItems = await dbTreeDataProvider.getChildren();
+
+    expect(dbTreeItems).toBeTruthy();
+    const items = dbTreeItems!;
+
+    const remoteRootNode = items[0];
+    expect(remoteRootNode.dbItem).toBeTruthy();
+    expect(remoteRootNode.dbItem?.kind).toEqual(DbItemKind.RootRemote);
+
+    const list1 = remoteRootNode.children.find(
+      (c) =>
+        c.dbItem?.kind === DbItemKind.RemoteUserDefinedList &&
+        c.dbItem?.listName === "my-list-1",
+    );
+    const list2 = remoteRootNode.children.find(
+      (c) =>
+        c.dbItem?.kind === DbItemKind.RemoteUserDefinedList &&
+        c.dbItem?.listName === "my-list-2",
+    );
+
+    expect(list1).toBeTruthy();
+    expect(list2).toBeTruthy();
+    expect(isTreeViewItemSelectable(list1!)).toBeTruthy();
+    expect(isTreeViewItemSelected(list2!)).toBeTruthy();
+  });
+
+  it("should mark selected remote db inside list as selected", async () => {
```

Comment:
Is it worth testing local cases here too? (I'm guessing no, given the shift in priorities, but just double-checking!)

Reply:
I thought since this is an integration test it's not worth covering every case there is so just picked some.

Reply:
Makes sense!